typescript/infra/scripts/funding/reclaim-from-igp.ts

```diff
@@ -1,53 +1,250 @@
-import { BigNumber } from 'ethers';
+import chalk from 'chalk';
+import { formatEther, parseEther } from 'ethers/lib/utils.js';
 
 import { HyperlaneIgp } from '@hyperlane-xyz/sdk';
-import { objMap, promiseObjAll } from '@hyperlane-xyz/utils';
+import {
+  isZeroishAddress,
+  objMap,
+  promiseObjAll,
+  rootLogger,
+} from '@hyperlane-xyz/utils';
 
 import { getEnvAddresses } from '../../config/registry.js';
-import { getArgs } from '../agent-utils.js';
+import { getKeyFunderConfig } from '../../src/funding/key-funder.js';
+import { TurnkeyRole } from '../../src/roles.js';
+import { setTurnkeySignerForEvmChains } from '../../src/utils/turnkey.js';
+import { getArgs, withChains } from '../agent-utils.js';
 import { getEnvironmentConfig } from '../core-utils.js';
 
-// Some arbitrary threshold for now
-const RECLAIM_BALANCE_THRESHOLD = BigNumber.from(10).pow(17);
+function withForce<T>(args: any) {
+  return args
+    .describe('force', 'Force claim even if below threshold')
+    .boolean('force')
+    .alias('f', 'force')
+    .default('force', false);
+}
+
+const ReclaimStatus = {
+  SUCCESS: '🟢',
+  BELOW_THRESHOLD: '🔵',
+  INSUFFICIENT_FOR_GAS: '🟡',
+  NO_GAS_PRICE: '🟡',
+  ERROR: '🔴',
+} as const;
+type ReclaimStatus = (typeof ReclaimStatus)[keyof typeof ReclaimStatus];
+
+interface ReclaimResult {
+  chain: string;
+  balance: string;
+  threshold: string;
+  status: ReclaimStatus;
+}
+
+// Format to 5 significant figures
+function formatTo5SF(value: string): string {
+  const num = parseFloat(value);
+  if (num === 0) return '0';
+  return num.toPrecision(5);
```

Comment:
nit: shouldn't we also check that the provided string does not end up being `Nan`?